lib/umwelt/client/writer.ex

```diff
@@ -0,0 +1,41 @@
+defmodule Umwelt.Client.Writer do
+  @moduledoc "File writer task"
+
+  use Task
+  require Logger
+
+  alias Umwelt.Client
+
+  def start_link(module) do
+    Task.start_link(__MODULE__, :run, [module])
```

Comment:
`Task.Supervisor` would probably help to make sure all tasks are finished normally.